sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalysisContext.java

```diff
@@ -47,6 +48,14 @@ public boolean hasAggregates() {
         return hasAggregates;
     }
 
+    void forceReturnMultipleRows(boolean forceReturnMultipleRows) {
+        this.forceReturnMultipleRows = forceReturnMultipleRows;
```

Comment:
I don't understand why we have to introduce a second facility for this. Can't we use the existing `isArrayComparisonChild` functionality, but adapt it to fit both cases?